fabric-rendering-v1/src/client/java/net/fabricmc/fabric/mixin/client/rendering/WorldRendererMixin.java

```diff
@@ -16,18 +16,164 @@
+	private void onPushBlockEntityRenders(WorldRenderer instance, MatrixStack matrices, WorldRenderState worldRenderState, OrderedRenderCommandQueueImpl commandQueue, Operation<Void> original) {
+		original.call(instance, matrices, worldRenderState, commandQueue);
+		WorldRenderEvents.AFTER_SUBMIT_ENTITY_COMMANDS.invoker().afterSubmitEntityCommands(renderContext);
+	}
+
+	@WrapOperation(method = "method_62214",
+			slice = @Slice(from = @At(value = "INVOKE", target = "Lnet/minecraft/client/render/command/RenderDispatcher;render()V")),
+			at = @At(value = "INVOKE", target = "Lnet/minecraft/client/render/VertexConsumerProvider$Immediate;drawCurrentLayer()V")
+	)
+	private void onEntityRender(VertexConsumerProvider.Immediate instance, Operation<Void> original) {
+		WorldRenderEvents.BEFORE_ENTITY_RENDER.invoker().beforeEntityRender(renderContext);
+		original.call(instance);
+		WorldRenderEvents.AFTER_ENTITY_RENDER.invoker().afterEntityRender(renderContext);
+	}
+
+	@WrapOperation(method = "method_62214", at = @At(value = "INVOKE", target = "Lnet/minecraft/client/render/debug/DebugRenderer;render(Lnet/minecraft/client/util/math/MatrixStack;Lnet/minecraft/client/render/Frustum;Lnet/minecraft/client/render/VertexConsumerProvider$Immediate;DDDZ)V"))
+	private void onDebugRender(DebugRenderer instance, MatrixStack matrices, Frustum frustum, VertexConsumerProvider.Immediate consumers, double cameraX, double cameraY, double cameraZ, boolean lateDebug, Operation<Void> original) {
+		original.call(instance, matrices, frustum, consumers, cameraX, cameraY, cameraZ, lateDebug);
+		WorldRenderEvents.AFTER_DEBUG_RENDER.invoker().afterDebugRender(renderContext);
+	}
+
+	@WrapOperation(method = "method_62214",
+			slice = @Slice(from = @At(value = "INVOKE_STRING", target = "Lnet/minecraft/util/profiler/Profiler;push(Ljava/lang/String;)V", args = "ldc=translucent")),
+			at = @At(value = "INVOKE", target = "Lnet/minecraft/client/render/SectionRenderState;renderSection(Lnet/minecraft/client/render/BlockRenderLayerGroup;)V")
+	)
+	private void onTranslucentRender(SectionRenderState instance, BlockRenderLayerGroup group, Operation<Void> original) {
+		original.call(instance, group);
+		WorldRenderEvents.AFTER_TRANSLUCENT_RENDER.invoker().afterTranslucentRender(renderContext);
+	}
+
+	@Inject(method = "renderTargetBlockOutline", at = @At(value = "FIELD", target = "Lnet/minecraft/client/render/state/CameraRenderState;pos:Lnet/minecraft/util/math/Vec3d;"), cancellable = true)
+	private void onDrawBlockOutline(VertexConsumerProvider.Immediate consumers, MatrixStack matrices, boolean bl, WorldRenderState worldRenderState, CallbackInfo ci) {
+		if (!WorldRenderEvents.BEFORE_BLOCK_OUTLINE_RENDER.invoker().beforeBlockOutlineRender(renderContext)) {
+			consumers.drawCurrentLayer();
+			ci.cancel();
+		}
+	}
+
+	@Inject(method = "method_62214", at = @At("RETURN"))
```

Comment:
You probably want this just before the last `draw`, not after
```suggestion
	@Inject(method = "method_62214", at = @At(value = "INVOKE:LAST", target = "Lnet/minecraft/client/render/VertexConsumerProvider$Immediate;draw()V"))
```